### PR TITLE
Auto-TypeSpec Part 1: parity

### PR DESCRIPTION
**Pull request by trrwilson**
I've been working on a very narrowly scoped OpenAPI -> .tsp ingestion tool, located here:
https://github.com/trrwilson/OpenApiToTsp

The objective of that tool is to generate the .tsp models from OpenAI's OpenAPI document with zero hand-editing; modifications are accomplished via a .yaml config file.

Every models.tsp file in this PR was entirely generated by a combination of that tool and `tsp format`; no hand editing was done outside of matched operations removal in Assistants v2 (the tool currently doesn't touch operations), a convenience alias move into a "custom.tsp" file, and import cleanup.

The client was then regenerated and the couple of custom code changes needed (to handle BinaryData use with the loss of a custom type; we could easily reintroduce this using the same pattern as above, if we so desire) and all existing tests pass.

### CODE REVIEW

File: images/models.tsp [hunk 1]
```diff
@@ -1,151 +1,161 @@
...
+  user?: string;
 }
 
 model CreateImageVariationRequest {
   /**
    * The image to use as the basis for the variation(s). Must be a valid PNG file, less than 4MB,
    * and square.
    */
-  @encode("binary")
-  image: bytes;
```
Comment (joseharriaga) [inline images/models.tsp:107]: Should we keep this as bytes?

File: completions/models.tsp [hunk 2]
```diff
@@ -45,130 +45,107 @@ model CreateCompletionRequest {
...
+  stop?: string | string[] | null;
+
+  @doc("""
+  Whether to stream back partial progress. If set, tokens will be sent as data-only [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#Event_stream_format) as they become available, with the stream terminated by a `data: [DONE]` message. [Example Python code](https://cookbook.openai.com/examples/how_to_stream_completions).
+  """)
   stream?: boolean | null = false;
 
   /** The suffix that comes after a completion of inserted text. */
-  suffix?: string | null = null;
+  suffix?: string | null;
```
Comment (joseharriaga) [inline completions/models.tsp:124]: Should default to null

File: completions/models.tsp [hunk 3]
```diff
@@ -179,60 +156,48 @@ model CreateCompletionResponse {
...
+
+      tokens?: string[];
+
+      top_logprobs?: Record<float32>[];
+    } | null;
+
+    text: string;
   }[];
 
+  // Tool customization: _at and created are Unix encoded utcDateTime
```
Comment (joseharriaga) [inline completions/models.tsp:183]: What does "_at" mean?

File: threads/models.tsp [hunk 4]
```diff
@@ -38,18 +45,34 @@ model ThreadObject {
...
+   * additional information about the object in a structured format. Keys can be a maximum of 64
+   * characters long and values can be a maxium of 512 characters long.
+   */
   @extension("x-oaiTypeLabel", "map")
   metadata: Record<string> | null;
-}
\ No newline at end of file
+}
+
+model ListThreadsResponse {
```
Comment (joseharriaga) [inline threads/models.tsp:67]: Funny. Is there an undocumented "list threads" endpoint?

File: models/models.tsp [hunk 5]
```diff
@@ -1,23 +1,32 @@
...
   object: "list";
+
   data: Model[];
 }
 
 model DeleteModelResponse {
   id: string;
+
   deleted: boolean;
-  object: "model"; // TODO: This is just a string in the OpenAPI spec, not an enum.
```
Comment (joseharriaga) [inline models/models.tsp:13]: We should keep this fix.

File: assistants/models.tsp [hunk 6]
```diff
@@ -93,32 +97,41 @@ model ModifyAssistantRequest {
...
-   * use. Useful for tools like `retrieval` and `code_interpreter` that can access files.
-   */
+  @doc("""
+  A [File](/docs/api-reference/files) ID (with `purpose="assistants"`) that the assistant should use. Useful for tools like `retrieval` and `code_interpreter` that can access files.
+  """)
   file_id: string;
 }
 
 model ListAssistantsResponse {
-  object: "list";
```
Comment (joseharriaga) [inline assistants/models.tsp:104]: We should keep this fix.

File: assistants/models.tsp [hunk 7]
```diff
@@ -93,32 +97,41 @@ model ModifyAssistantRequest {
...
 model DeleteAssistantResponse {
   id: string;
+
   deleted: boolean;
+
   object: "assistant.deleted";
 }
 
 model ListAssistantFilesResponse {
-  object: "list";
```
Comment (joseharriaga) [inline assistants/models.tsp:118]: We should keep this fix.

File: messages/models.tsp [hunk 8]
```diff
@@ -1,82 +1,106 @@
...
+   * Set of 16 key-value pairs that can be attached to an object. This can be useful for storing
+   * additional information about the object in a structured format. Keys can be a maximum of 64
+   * characters long and values can be a maxium of 512 characters long.
+   */
   @extension("x-oaiTypeLabel", "map")
   metadata?: Record<string> | null;
 }
 
 model ListMessagesResponse {
-  object: "list";
```
Comment (joseharriaga) [inline messages/models.tsp:46]: We should keep this fix.

File: messages/models.tsp [hunk 9]
```diff
@@ -1,82 +1,106 @@
...
+
   first_id: string;
+
   last_id: string;
+
   has_more: boolean;
 }
 
 model ListMessageFilesResponse {
-    object: "list";
```
Comment (joseharriaga) [inline messages/models.tsp:54]: We should keep this fix.

File: runs/models.tsp [hunk 10]
```diff
@@ -103,142 +115,121 @@ model SubmitToolOutputsRunRequest {
 }
 
 model ListRunsResponse {
-  object: "list";
```
Comment (joseharriaga) [inline runs/models.tsp:106]: We should keep this fix.

File: runs/models.tsp [hunk 11]
```diff
@@ -103,142 +115,121 @@ model SubmitToolOutputsRunRequest {
...
+
   first_id: string;
+
   last_id: string;
+
   has_more: boolean;
 }
 
 model ListRunStepsResponse {
-  object: "list";
```
Comment (joseharriaga) [inline runs/models.tsp:114]: We should keep this fix.

File: runs/models.tsp [hunk 12]
```diff
@@ -389,17 +381,20 @@ model RunObject {
   @extension("x-oaiTypeLabel", "map")
   metadata: Record<string> | null;
 
-  usage: RunCompletionUsage | null;
+  usage: RunCompletionUsage;
```
Comment (joseharriaga) [inline runs/models.tsp:384]: This is nullable.

File: runs/models.tsp [hunk 13]
```diff
@@ -461,7 +466,7 @@ model RunStepObject {
   @extension("x-oaiTypeLabel", "map")
   metadata: Record<string> | null;
 
-  usage: RunStepCompletionUsage | null;
+  usage: RunStepCompletionUsage;
```
Comment (joseharriaga) [inline runs/models.tsp:469]: This is nullable.

File: chat/models.tsp [hunk 14]
```diff
@@ -1,168 +1,150 @@
...
+   * 
    * Accepts a JSON object that maps tokens (specified by their token ID in the tokenizer) to an
    * associated bias value from -100 to 100. Mathematically, the bias is added to the logits
    * generated by the model prior to sampling. The exact effect will vary per model, but values
    * between -1 and 1 should decrease or increase likelihood of selection; values like -100 or 100
    * should result in a ban or exclusive selection of the relevant token.
    */
   @extension("x-oaiTypeLabel", "map")
-  logit_bias?: Record<safeint> | null = null;
+  logit_bias?: Record<int32> | null;
```
Comment (joseharriaga) [inline chat/models.tsp:53]: This defaults to null

File: chat/models.tsp [hunk 15]
```diff
@@ -1,168 +1,150 @@
...
+  @minValue(-9223372036854775808)
   @maxValue(9223372036854775807)
-  seed?: safeint | null;
+  seed?: int64 | null;
 
-  // TODO: Consider inlining when https://github.com/microsoft/typespec/issues/2356 is resolved
-  // https://github.com/microsoft/typespec/issues/2355
   /** Up to 4 sequences where the API will stop generating further tokens. */
-  stop?: Stop | null = null;
+  stop?: string | string[];
```
Comment (joseharriaga) [inline chat/models.tsp:119]: This defaults to null

File: chat/models.tsp [hunk 16]
```diff
@@ -204,149 +183,97 @@ model CreateChatCompletionResponse {
...
-   * passing in images. You can pass multiple images by adding multiple `image_url` content parts.
-   * Image input is only supported when using the `gpt-4-visual-preview` model.
-   */
-  ChatCompletionRequestMessageContentParts,
-};
+`none` is the default when no functions are present. `auto` is the default if functions are present.
+""")
+@extension("x-oaiExpandable", true)
+union ChatCompletionToolChoiceOption {
+"none" | "auto",
```
Comment (joseharriaga) [inline chat/models.tsp:265]: Should it be a comma instead of a pipe between "none" and "auto":

```
"none",
"auto",
ChatCompletionNameToolChoice
```

File: chat/models.tsp [hunk 17]
```diff
@@ -204,149 +183,97 @@ model CreateChatCompletionResponse {
...
-model ChatCompletionRequestMessageContentParts is ChatCompletionRequestMessageContentPart[];
 
-@oneOf
 @extension("x-oaiExpandable", true)
 union ChatCompletionRequestMessageContentPart {
-  ChatCompletionRequestMessageContentPartText,
-  ChatCompletionRequestMessageContentPartImage,
+ChatCompletionRequestMessageContentPartText,
+ChatCompletionRequestMessageContentPartImage
 }
```
Comment (joseharriaga) [inline chat/models.tsp:274]: nit: Unions are missing indentation.

File: chat/models.tsp [hunk 18]
```diff
@@ -454,11 +384,9 @@ model ChatCompletionRequestAssistantMessage {
 
   tool_calls?: ChatCompletionMessageToolCalls;
 
-  /**
-   * Deprecated and replaced by `tool_calls`. The name and arguments of a function that should be
-   * called, as generated by the model.
-   */
-  #deprecated "deprecated"
```
Comment (joseharriaga) [inline chat/models.tsp:461]: Should we keep the #deprecated annotation?

File: chat/models.tsp [hunk 19]
```diff
@@ -504,8 +485,9 @@ model ChatCompletionResponseMessage {
   /** The role of the author of this message. */
   role: "assistant";
 
-  /** Deprecated and replaced by `tool_calls`. The name and arguments of a function that should be called, as generated by the model. */
-  #deprecated "deprecated"
```
Comment (joseharriaga) [inline chat/models.tsp:508]: Should we keep the #deprecated annotation?

File: chat/models.tsp [hunk 20]
```diff
@@ -520,54 +502,46 @@ model ChatCompletionResponseMessage {
...
- */
+@doc("""
+Specifying a particular function via `{"name": "my_function"}` forces the model to call that function.
+""")
 model ChatCompletionFunctionCallOption {
   /** The name of the function to call. */
   name: string;
 }
 
-#deprecated "deprecated"
```
Comment (joseharriaga) [inline chat/models.tsp:570]: #deprecated annotation?

File: audio/models.tsp [hunk 21]
```diff
@@ -142,15 +137,12 @@ model CreateTranscriptionResponseJson {
...
-  /** The task label. */
-  task: "transcribe";
-
   /** The language of the input audio. */
   language: string;
 
+  // Tool customization: timespans are encoded durations
   /** The duration of the input audio. */
-  @encode("seconds", float64)
+  @encode("seconds", float)
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply audio/models.tsp:145]: Pending our discussion, I'll standardize everything to `float32` across the board.

File: fine-tuning/models.tsp [hunk 22]
```diff
@@ -138,18 +128,17 @@ model FineTuningJob {
   finished_at: utcDateTime | null;
 
   /**
-   * The hyperparameters used for the fine-tuning job. See the
-   * [fine-tuning guide](/docs/guides/fine-tuning) for more details.
+   * The hyperparameters used for the fine-tuning job. See the [fine-tuning
+   * guide](/docs/guides/fine-tuning) for more details.
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply fine-tuning/models.tsp:132]: I'll remove the line breaks entirely for now.

I don't think that consumption of these comment blocks should be opinionated about line breaks, but that's clearly the case in some other circumstances, too; it's very human-readability-ugly to have to rely on an editor wrapping to read long lines, but that'll at least be functional.

File: completions/models.tsp (hunk 3 above)
Comment (trrwilson) [reply completions/models.tsp:183]: Revised the rule's remark:
```
  // Tool customization: 'created' and fields ending in '_at' are Unix encoded utcDateTime
```

File: threads/models.tsp (hunk 4 above)
Comment (trrwilson) [reply threads/models.tsp:67]: Oh, that's interesting! Yes, I believe there must be; it's very strange that the operation doesn't show up anywhere. We should look into that, but for this purpose I think we should keep parity.

File: models/models.tsp (hunk 5 above)
Comment (trrwilson) [reply models/models.tsp:13]: Is that still the case?
```
    Model:
      title: Model
      description: Describes an OpenAI model offering that can be used with the API.
      properties:
        id:
          type: string
          description: The model identifier, which can be referenced in the API endpoints.
        created:
          type: integer
          description: The Unix timestamp (in seconds) when the model was created.
        object:
          type: string
          description: The object type, which is always "model".
          enum: [model]
```

File: chat/models.tsp (hunk 16 above)
Comment (trrwilson) [reply chat/models.tsp:265]: I believe they're equivalent:
https://typespec.io/playground?c=dW5pb24gRm9vIHsNCiAgImhlbGxvIiB8ICJ3b3JsZCINCn0NCg0KxidCYXLNJyzFDMkp&e=%40typespec%2Fopenapi3&options=%7B%7D

File: audio/models.tsp [hunk 23]
```diff
@@ -168,91 +160,76 @@ model CreateTranslationResponseJson {
 }
 
 model CreateTranslationResponseVerboseJson {
-  // TODO: This is not defined in the OpenAPI spec.
-  /** The task label. */
-  task: "translate";
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply audio/models.tsp:173]: Restored.

File: chat/models.tsp (hunk 16 above)
Comment (trrwilson) [reply chat/models.tsp]: (we can update this later if we don't like the format, though)

Comment (trrwilson) [reply chat/models.tsp]: As written, it's a more literal translation of the `oneOf` schema:
```
      oneOf:
        - type: string
          description: >
            `none` means the model will not call a function and instead generates a message.
            `auto` means the model can pick between generating a message or calling a function.
          enum: [none, auto]
        - $ref: "#/components/schemas/ChatCompletionNamedToolChoice"
```

File: chat/models.tsp (hunk 18 above)
Comment (trrwilson) [reply chat/models.tsp:461]: Systematically added.

File: chat/models.tsp (hunk 17 above)
Comment (trrwilson) [reply chat/models.tsp]: I must've missed running `tsp format` before the last push! Should be addressed, and several other things are probably more readable, too. Sorry about that!

File: fine-tuning/models.tsp [hunk 24]
```diff
@@ -10,59 +15,59 @@ model CreateFineTuningJobRequest {
...
-   * See the [fine-tuning guide](/docs/guides/fine-tuning) for more details.
-   */
+  @doc("""
+  A string of up to 18 characters that will be added to your fine-tuned model name.
+  
+  For example, a `suffix` of "custom-model-name" would produce a model name like `ft:gpt-3.5-turbo:openai:custom-model-name:7p4lURel`.
+  """)
+  @minLength(1)
+  @maxLength(40)
+  suffix?: string | null;
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply fine-tuning/models.tsp]: Should be systematically addressed. That was a tricky one; I was storing `DefaultValue` as an `object` and it thus had no distinction between "no default" and "null default."

File: assistants/models.tsp (hunk 7 above)
Comment (trrwilson) [reply assistants/models.tsp:118]: Note: I applied a potentially dangerous catch-all rule for "List.*Response.object". I don't see any conflict in doing it that way and that helps with forward application, but it's worth ensuring that's the case.

File: .dotnet/src/OpenAI.csproj [hunk 25]
```diff
@@ -4,12 +4,13 @@
     <AssemblyTitle>SDK Code Generation OpenAI</AssemblyTitle>
     <Version>1.0.0-beta.1</Version>
     <PackageTags>OpenAI</PackageTags>
-    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
```
Comment (joseharriaga) [inline .dotnet/src/OpenAI.csproj:7]: We should keep targeting .NET 6.

File: audio/models.tsp [hunk 26]
```diff
@@ -138,19 +107,14 @@ model CreateTranscriptionResponseJson {
...
 }
 
-/**
- * Represents a verbose json transcription response returned by model, based on the provided input.
- */
+/** Represents a verbose json transcription response returned by model, based on the provided input. */
 model CreateTranscriptionResponseVerboseJson {
-  // TODO: This is not defined in the OpenAPI spec.
-  /** The task label. */
-  task: "transcribe";
```
Comment (joseharriaga) [inline audio/models.tsp:147]: We should restore this one too.

File: .dotnet/src/Custom/Audio/AudioTranscription.cs [hunk 27]
```diff
@@ -9,7 +9,7 @@ namespace OpenAI.Audio;
 public partial class AudioTranscription
 {
     // CUSTOM: Made private. This property does not add value in the context of a strongly-typed class.
-    private CreateTranscriptionResponseVerboseJsonTask Task { get; } = CreateTranscriptionResponseVerboseJsonTask.Transcribe;
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Audio/AudioTranscription.cs:12]: We can undo this change once the property is manually restored in the TypeSpec.

File: .dotnet/src/Custom/Audio/AudioTranslation.cs [hunk 28]
```diff
@@ -9,7 +9,7 @@ namespace OpenAI.Audio;
 public partial class AudioTranslation
 {
     // CUSTOM: Made private. This property does not add value in the context of a strongly-typed class.
-    private CreateTranslationResponseVerboseJsonTask Task { get; } = CreateTranslationResponseVerboseJsonTask.Translate;
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Audio/AudioTranslation.cs:12]: We can undo this change once the property is manually restored in the TypeSpec.

File: .dotnet/src/Custom/Models/DeleteModelResponse.cs [hunk 29]
```diff
@@ -5,5 +5,5 @@ public partial class DeleteModelResponse
 {
     // CUSTOM: Made private. This property does not add value in the context of a strongly-typed class.
     /// <summary> Gets the object. </summary>
-    private DeleteModelResponseObject Object { get; } = DeleteModelResponseObject.Model;
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Models/DeleteModelResponse.cs:8]: We can undo this change once the property is manually fixed in the TypeSpec.

File: .dotnet/src/Custom/Embeddings/EmbeddingTokenUsage.cs [hunk 30]
```diff
@@ -1,10 +1,16 @@
...
     /// <summary> The number of tokens used by the input prompts. </summary>
     [CodeGenMember("PromptTokens")]
-    public long InputTokens { get; }
+    public int InputTokens { get; }
+
+    // CUSTOM: Convenience member added for completeness.
+    /// <summary>
+    /// The number of tokens used by embedding outputs.
+    /// </summary>
+    public int OutputTokens => TotalTokens - InputTokens;
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Embeddings/EmbeddingTokenUsage.cs:15]: I wonder if it would be advisable to not add custom properties to this class, mostly because I'm not sure how much value this approach to measuring usage is in the first place. That being said, I don't feel strongly.

File: .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs [hunk 31]
```diff
@@ -26,7 +26,7 @@ void IJsonModel<AudioTranslation>.Write(Utf8JsonWriter writer, ModelReaderWriter
             writer.WritePropertyName("language"u8);
             writer.WriteStringValue(Language);
             writer.WritePropertyName("duration"u8);
-            writer.WriteNumberValue(Convert.ToInt32(Duration.Value.ToString("%s")));
+            writer.WriteNumberValue(Convert.ToDouble(Duration.Value.ToString("s\\.fff")));
```
Comment (joseharriaga) [inline .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs:29]: This change looks correct but... How did it happen? Are you using a newer version of the generator? Does this mean that the custom serialization fix for TimeSpans encoded as seconds is no longer needed?

File: .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs [hunk 32]
```diff
@@ -26,7 +26,7 @@ void IJsonModel<AudioTranslation>.Write(Utf8JsonWriter writer, ModelReaderWriter
             writer.WritePropertyName("language"u8);
             writer.WriteStringValue(Language);
             writer.WritePropertyName("duration"u8);
```
Comment (joseharriaga) [inline .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs:28]: Unrelated to your PR, this code crashes because Duration was customized to be made nullable but this code doesn't account for the possibility of a null value... Sent a message to the generator team.

File: .dotnet/src/Generated/Models/CreateTranscriptionResponseVerboseJsonTask.cs [hunk 33]
```diff
@@ -1,45 +0,0 @@
-// <auto-generated/>
```
Comment (joseharriaga) [inline .dotnet/src/Generated/Models/CreateTranscriptionResponseVerboseJsonTask.cs:1]: This one should be restored.

File: .dotnet/src/Generated/Models/DeleteModelResponse.Serialization.cs [hunk 34]
```diff
@@ -26,7 +26,7 @@ void IJsonModel<DeleteModelResponse>.Write(Utf8JsonWriter writer, ModelReaderWri
             writer.WritePropertyName("deleted"u8);
             writer.WriteBooleanValue(Deleted);
             writer.WritePropertyName("object"u8);
-            writer.WriteStringValue(Object.ToString());
```
Comment (joseharriaga) [inline .dotnet/src/Generated/Models/DeleteModelResponse.Serialization.cs:29]: This one will be restored after the TypeSpec fix.

File: .dotnet/src/Custom/Embeddings/Embedding.cs [hunk 35]
```diff
@@ -59,34 +59,15 @@ public partial class Embedding
...
 
     // CUSTOM: Added logic to handle additional custom properties.
-    /// <summary> Initializes a new instance of <see cref="OpenAI.Embeddings.Embedding"/>. </summary>
-    /// <param name="index"> The index of the embedding in the list of embeddings. </param>
-    /// <param name="embeddingProperty">
-    /// The embedding vector, which is a list of floats. The length of vector depends on the model as
-    /// listed in the [embedding guide](/docs/guides/embeddings).
-    /// </param>
-    /// <exception cref="ArgumentNullException"> <paramref name="embeddingProperty"/> is null. </exception>
-    internal Embedding(long index, BinaryData embeddingProperty)
```
Comment (joseharriaga) [inline .dotnet/src/Custom/Embeddings/Embedding.cs:69]: By deleting this constructor, it has moved back to the generated layer. But the constrcutor in the generated layer doesn't set the custom Vector property. I recommend restoring this constructor (and changing the long parameter to an int parameter) to ensure that all constructors work correctly. Alternatively, this constructor can be suppressed with CodeGenSuppress.

File: .dotnet/src/Generated/Models/Embedding.cs [hunk 36]
```diff
@@ -42,12 +42,24 @@ public partial class Embedding
         /// </summary>
         private IDictionary<string, BinaryData> _serializedAdditionalRawData;
 
+        /// <summary> Initializes a new instance of <see cref="Embedding"/>. </summary>
+        /// <param name="index"> The index of the embedding in the list of embeddings. </param>
+        /// <param name="embeddingProperty"> The embedding vector, which is a list of floats. The length of vector depends on the model as listed in the [embedding guide](/docs/guides/embeddings). </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="embeddingProperty"/> is null. </exception>
+        internal Embedding(int index, BinaryData embeddingProperty)
```
Comment (joseharriaga) [inline .dotnet/src/Generated/Models/Embedding.cs:49]: See my comment in the custom Embedding.cs file about this constructor.

File: .dotnet/src/OpenAI.csproj (hunk 25 above)
Comment (trrwilson) [reply .dotnet/src/OpenAI.csproj:7]: Oh, good catch! The `Update-Client` script keeps regenerating this file with this weird change, for some reason; I missed reverting it.

File: .dotnet/src/Custom/Embeddings/EmbeddingTokenUsage.cs (hunk 30 above)
Comment (trrwilson) [reply .dotnet/src/Custom/Embeddings/EmbeddingTokenUsage.cs:15]: Can always add it later. I'll remove it.

File: .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs (hunk 31 above)
Comment (trrwilson) [reply .dotnet/src/Generated/Models/AudioTranslation.Serialization.cs:29]: It must be -- that or a generator update since the customizations were added. It does appear that it's fixed, one way or another, and that the custom logic can go away!

File: .dotnet/src/Custom/Embeddings/Embedding.cs (hunk 35 above)
Comment (trrwilson) [reply .dotnet/src/Custom/Embeddings/Embedding.cs:69]: Opted for the suppression route since it's not being used.
